cpp/src/hash/managed_allocator.cuh

```diff
@@ -60,9 +54,9 @@ struct legacy_allocator {
 
       template <class U> constexpr legacy_allocator(const legacy_allocator<U>&) noexcept {}

```

Comment:
I would like to rename this allocator. It's not "legacy", its default. This is the one we use in cuDF. It should also be updated to use a device_memory_resource.

Reply:
I renamed this allocator to `default_allocator` and changed it to use the default resource of RMM